local/node.go

```diff
@@ -38,27 +42,126 @@ type getConnFunc func(context.Context, node.Node) (net.Conn, error)
 // AvalancheGo binaries in tests
 type NodeProcess interface {
 	// Start this process
-	Start() error
+	// Returns error if not called after instantiation
+	// If process stops without a previous call to Stop(), send notification msg over given channel
+	Start(chan network.UnexpectedNodeStopMsg) error
 	// Send a SIGTERM to this process
-	Stop() error
+	// If ctx is cancelled, send SIGKILL to this process and descendants
+	// Returns error if called before Start
+	// Returns nil if the process is already stopping/stopped
+	Stop(ctx context.Context) error
 	// Returns when the process finishes exiting
+	// Returns error if called before Start
+	// Returns nil if already waited for the process
 	Wait() error
+	// Returns true if the process is executing
+	// Returns false if the process has been stopped
+	// Returns false if the process has not been started
+	Alive() bool
 }
 
 type nodeProcessImpl struct {
-	cmd *exec.Cmd
+	name string
+	lock sync.RWMutex
+	cmd  *exec.Cmd
+	// to notify Wait() on process stop, and give wait return
+	waitReturnCh chan error
+	// to notify user of not asked process stops
+	unexpectedStopCh chan network.UnexpectedNodeStopMsg
+	// maintains process state Initial/Started/Stopping/Stopped/Waited
+	state int
+	// to notify SIGKILL goroutine of process end
+	closeOnStop chan struct{}
 }
 
-func (p *nodeProcessImpl) Start() error {
-	return p.cmd.Start()
+const (
+	Initial = iota
+	Started
+	Stopping
+	Stopped
+	Waited
```

Comment:
What does state `Waited` mean? We set the state to `Waited` after `Wait` returns, but we also set the state to `Stopped` in the goroutine inside `Start`, so it it's possible for the state to be either `Waited` or `Stopped` after the node process stops.

Reply:
removed state Waited